HomesEngland/UseCase/SearchAsset/Impl/SearchAssetUseCase.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using HomesEngland.Domain;
+using HomesEngland.Exception;
+using HomesEngland.Gateway.Assets;
+using HomesEngland.UseCase.GetAsset;
+using HomesEngland.UseCase.GetAsset.Models;
+using HomesEngland.UseCase.SearchAsset.Models;
+using Infrastructure.Api.Exceptions;
+
+namespace HomesEngland.UseCase.SearchAsset.Impl
+{
+    public class SearchAssetUseCase : ISearchAssetUseCase
+    {
+        private readonly IAssetSearcher _assetSearcher;
+
+        public SearchAssetUseCase(IAssetSearcher assetSearcher)
+        {
+            _assetSearcher = assetSearcher;
+        }
+
+        public async Task<SearchAssetResponse> ExecuteAsync(SearchAssetRequest request, CancellationToken cancellationToken)
+        {
+            if (request == null)
+                throw new BadRequestException();
+            var validationResponse = request.Validate(request);
+            if (!validationResponse.IsValid)
+                throw new BadRequestException(validationResponse);
+
+            IAssetSearch assetSearch = new AssetSearch
+            {
+                SchemeId = request.SchemeId
+            };
+
+            var assets = await _assetSearcher.Search(assetSearch, cancellationToken).ConfigureAwait(false);
+            if(assets == null)
+                throw new AssetNotFoundException();
+
+            var response = new SearchAssetResponse
+            {
+                Assets = assets?.Select(s=> new AssetOutputModel(s)).ToList()

```

Comment:
```suggestion
                Assets = assets.Select(s=> new AssetOutputModel(s)).ToList()
```

We already know that it's not null at this point